pkg/edgeview/src/basics.go

```diff
@@ -585,6 +586,22 @@ func listRecursiveFiles(path, pattern string) ([]string, error) {
 	return jfiles, nil
 }
 
+// get the cluster api-server IP and port from kubeconfig file
+func getKubeServerIPandPort(kubeConfigFile string) (string, error) {
+	content, err := os.ReadFile(kubeConfigFile)
+	if err != nil {
+		return "", err
+	}
+
+	regex := regexp.MustCompile(`\s+server: https://([^ ]+)`)
```

Comment:
@christoph-zededa yes, that's true, it will match anything. But we know this is the file we created from k3s cluster setup, and we know there is this pattern which we need to match. So, this is not a general regexp trying to match any file in general case.

Reply:
and this format has been around since start of kubernetes api, it will be unlikely being changed, and we'll get an error for not finding the ip and port immediately if they do change for some reason.

Reply:
> and we'll get an error for not finding the ip and port immediately

Immediately? I don't see any test that would fail and we would get notified.

Reply:
Is it really that complicated to add such a test? Also I fear systest team only does tests for the whole system, not just component/unit tests.

Reply:
well, this kubeconfig file is dynamically generated on the EVE device, I doubt unit test is useful.

Reply:
How is it created?

Reply:
this one, after k3s install on the device, the k3s.yaml file is generated when first launching the k3s, in kube container's /etc/rancher/k3s/k3s.yaml, but we create our own 'debugginguser', and we copy the server part of the lines from the k3s.yaml unchanged, and adding our 'debugginguser' credentials and certificates into /run/.kube/k3s/user.yaml file. the 'server: https://' line is in the server part of the file

Reply:
hmm, okay, seems it cannot be tested easily then.

Resolved.